collector/Makefile

```diff
@@ -4,8 +4,7 @@ BASE_SPACE:=$(shell pwd)
 BUILD_SPACE:=$(BASE_SPACE)/build
 BUCKET_NAME:=lambda-artifacts-$(shell dd if=/dev/random bs=8 count=1 2>/dev/null | od -An -tx1 | tr -d ' \t\n')
 LAYER_NAME:=otel-collector
-
-VERSION=$(shell cat VERSION)
+VERSION=$(shell curl -s https://raw.githubusercontent.com/open-telemetry/opentelemetry-collector-contrib/main/versions.yaml | grep 'version:' | head -1 | awk '{print $$2}')
```

Comment:
What about grepping `/collector/go.mod` directly for `go.opentelemetry.io/collector ` (with space at end)?

Reply:
We can certainly switch to that approach.

Reply:
@tylerbenson to confirm, you mean getting [this line ](https://github.com/open-telemetry/opentelemetry-lambda/blob/c55ad8742d9d4714a950fe4eaf11ed3246673577/collector/go.mod#L135) right?

Reply:
That's what I was thinking of... I'm concerned with the approach you proposed as it expects we're always on the latest release, but that's not necessarily the case.  Open to other ideas though.

Reply:
made changes